packages/back-end/src/api/features/listFeatures.ts

```diff
@@ -21,15 +21,17 @@ export const listFeatures = createApiRequestHandler(listFeaturesValidator)(
       ),
       req.query
     );
-
     return {
-      features: filtered.map((feature) =>
-        getApiFeatureObj({
-          feature,
-          organization: req.organization,
-          groupMap,
-          experimentMap,
-        })
+      features: await Promise.all(
```

Comment:
I typically don't like doing an unbounded number of Mongo queries in parallel like this.  If someone has a lot of features, this could be doing dozens or hundreds of queries

Reply:
would you suggest that we fetch all at once then assign during code